crates/noirc_abi/src/input_parser/cli.rs

```diff
@@ -0,0 +1,20 @@
+use crate::{errors::InputParserError, AbiType};
+
+use super::{parse_str_to_field, InputValue};
+
+impl InputValue {
+    pub fn try_from_cli_args(
+        value: String,
+        param_type: &AbiType,
+    ) -> Result<InputValue, InputParserError> {
+        let input_value = match param_type {
+            AbiType::String { .. } => InputValue::String(value),
+            AbiType::Field | AbiType::Integer { .. } => {
+                InputValue::Field(parse_str_to_field(&value)?)
+            }
+            _ => return Err(InputParserError::AbiTypeMismatch(param_type.clone())),
+        };
+        // TODO: support complex types
```

Comment:
Yeah, this is the hard part associated with this feature. I think we should have at least a good way to handle simple structs and arrays in this PR.

Ideally we'd have a not-so-nice way to handle complex structs as well (more for use by progarms interacting with nargo through the CLI rather than humans)